File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *temp=head;
        ListNode *temp2=head;
        int count=0;
        while(temp!=NULL){
            count++;
            temp=temp->next;
        }
        if(count==1 && n==1){
            return NULL;
        }
        if(count==n){
            return head->next;
        }
        ListNode *prev=head;
        int count2=0;
        while(temp2!=NULL){
            if(count2==(count-n) && temp2->next!=NULL){
                prev->next=prev->next->next;
                break;
            }else if(count2==(count-n) && temp2->next==NULL){
                prev->next=NULL;
            }else{
                prev=temp2;
            }
            temp2=temp2->next;
            count2++;
        }
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head.next==null && n==1){
            return null;
        }
        else if(head.next==null){
            return head;
        }
        ListNode temp=head;
        int c=0;
        while(temp!=null){
            c++;
            temp=temp.next;
        }
        temp=head;
        if(c==n){
            head=head.next;
            return head;
        }
        int x=c-n;
        for(int i=1;i<x;i++){
            temp=temp.next;
        }
        temp.next=temp.next.next;
        return head;
    }
}